File: python/min-moves/test_challenge.py
#! /usr/bin/env python
import unittest
from challenge import minMoves


class MinMovesTests(unittest.TestCase):
    def test_example1(self):
        self.assertEqual(10, minMoves("8051", "1199"))

    def test_example2(self):
        self.assertEqual(15, minMoves("000", "555"))

    def test_example3(self):
        self.assertEqual(4, minMoves("109", "990"))


if __name__ == '__main__':
    unittest.main()

File: ruby/longest-sorted/test_challenge.rb
#! /usr/bin/env ruby
require 'minitest/autorun'
require './challenge'


class TestChallenge < Minitest::Test
  def exec_test_case(sentence, expected)
    calculator = Challenge.new(sentence)
    result = calculator.longestSorted
    if expected.nil?
      assert_nil result
    else
      assert_equal expected, result
    end
  end

  def test_example1
    exec_test_case("The autumn leaves almost glow.", "almost")
  end

  def test_example1b
    exec_test_case("Almost as the autumn leaves ALMOST glow.", "almost")
  end

  def test_example1c
    exec_test_case("The autumn leaves glow almost.", "almost")
  end

  def test_example2
    exec_test_case("A cool sheep sleeps.", "")
  end
end
